File: NActiviti/Sys.Bpm.Engine.API/Engine/api/commands/IStartProcessInstanceCmd.cs
using System.Collections.Generic;

namespace Sys.Workflow.Services.Api.Commands
{
    /// <summary>
    /// 启动流程命令
    /// </summary>
    public interface IStartProcessInstanceCmd
    {
        /// <summary>
        /// 业务键值
        /// </summary>
        string BusinessKey { get; set; }

        /// <summary>
        /// 流程定义id
        /// </summary>
        string ProcessDefinitionId { get; set; }

        /// <summary>
        /// 主流程关联的业务键值
        /// </summary>
        string ProcessDefinitionBusinessKey { get; set; }

        /// <summary>
        /// 主流程id
        /// </summary>
        string ProcessDefinitionKey { get; set; }

        /// <summary>
        /// 流程实例名
        /// </summary>
        string ProcessInstanceName { get; set; }

        /// <summary>
        /// 流程名称
        /// </summary>
        string ProcessName { get; set; }

        /// <summary>
        /// 启动表单
        /// </summary>
        string StartForm { get; set; }

        /// <summary>
        /// 租户id
        /// </summary>
        string TenantId { get; set; }

        /// <summary>
        /// 流程变量
        /// </summary>
        WorkflowVariable Variables { get; set; }

        /// <summary>
        /// 流程启动节点id
        /// </summary>
        string InitialFlowElementId { get; set; }

        /// <summary>
        /// 使用消息触发流程
        /// </summary>
        string StartByMessage { get; set; }
    }
}

File: BpmnEditor/src/model/cmd/IStartProcessInstanceCmd.ts

export interface IStartProcessInstanceCmd {
  /// <summary>
  /// 
  /// </summary>
  processDefinitionKey?: string;

  /// <summary>
  /// 流程实例业务名称,方便查询显示.
  /// </summary>
  processInstanceName?: string;

  tenantId?: string;

  /// <summary>
  /// 流程定义id
  /// </summary>
  processDefinitionId?: string;

  /// <summary>
  /// 流程启动变量
  /// </summary>
  variables?: any;

  /// <summary>
  /// 业务键值,主要用来保存启动流程时的业务主键,可以是主键id，
  /// 也可以是表单编号,同一流程只要保持唯一性就可以了
  /// </summary>
  businessKey?: string;

  /// <summary>
  /// 启动表单,根据表单判断
  /// </summary>
  startForm?: string;
}
